### PR TITLE
History

### PR DESCRIPTION
**Pull request by JuvitaSaini09**
### Added History Page

 **Files to check**

- src/context/historyVideoContext/historyVideoContext.js
- src/pages/historyVideoPage/historyVideoPage.css
- src/pages/historyVideoPage/historyVideoPage.js
- src/util/apiCall.js

### CODE REVIEW

File: src/pages/historyVideoPage/historyVideoPage.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                      className="img-thumbnail"
+                      src={item.thumbnailUrl}
+                      alt={item.title}
+                    />
+                
+                    <h1 className="play">
+                      <i className="fas fa-play"></i>play
+                    </h1>
+                  </Link>
+                  <span onClick={()=>deleteVideoFromHistoryHandler(item)} className="historyDustbin"><i className="fas fa-trash-alt fa-2x"></i></span>
```
Comment (gauravsinhaweb) [inline src/pages/historyVideoPage/historyVideoPage.js]: suggestion: this onClick can can be done inside a function for readability and clean code.

**Review by gauravsinhaweb — APPROVED** [review]
code and UI look clean, I have mentioned a suggestion, you can merge it!

File: src/components/singleVideo/singleVideo.js [hunk 2]
```diff
@@ -1,17 +1,22 @@
 import React, { useState, useEffect } from "react";
 import { useDisLikedVideoContext } from "../../context/disLikedVideoContext/disLikedVideoContext";
+import { useHistoryVideoContext } from "../../context/historyVideoContext/historyVideoContext";
 import { useLikedVideoContext } from "../../context/likedVideoContext/likedVideoContext";
 import { useSingleVideo } from "../../context/singleVideoContext/singleVideoContext";
 import { addToList } from "../../images/allImages";
```
Comment (yaswanthmyneni) [inline src/components/singleVideo/singleVideo.js]: suggestion: export all contexts from index.js and import here

File: src/pages/historyVideoPage/historyVideoPage.css [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    position: absolute;
+    right:0;
+    color:var(--grey-dark);
+    background-color:var(--white);
+    padding:5px;
+    cursor:pointer;
+  }
+
+  .historyDustbin :hover{
+    color:rgb(80, 80, 80)
```
Comment (yaswanthmyneni) [inline src/pages/historyVideoPage/historyVideoPage.css]: changes: use css variables for colors everywhere

File: src/util/apiCall.js [hunk 4]
```diff
@@ -42,3 +42,69 @@ export async function deleteLikedVideoApi(singleVideo, likedVideoDispatch) {
...
+      },
+    });
+    if (response.status === 200) {
+      historyVideoDispatch({
+        type: "removeSingeleVideoFromHistory",
+        payload: response.data.history,
+      });
+    }
+  } catch (error) {
+    console.log(error);
```
Comment (yaswanthmyneni) [inline src/util/apiCall.js]: suggestion: use console.error for errors

File: src/pages/historyVideoPage/historyVideoPage.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            {historyVideoState === undefined || historyVideoState[0] === undefined ? <EmptyPage text="Empty History" /> :<>{historyVideoState.map((item) => {
+              return (
+                <div className="video-card">
+                  <Link onClick={() => setSingleVideo(item)} to="/video">
+                    <img
+                      className="img-thumbnail"
+                      src={item.thumbnailUrl}
+                      alt={item.title}
+                    />
+
```
Comment (yaswanthmyneni) [inline src/pages/historyVideoPage/historyVideoPage.js]: suggestion: can move this to separate component

**Review by yaswanthmyneni — APPROVED** [review]
you can merge after changes

File: src/util/apiCall.js (hunk 4 above)
Comment (JuvitaSaini09) [reply src/util/apiCall.js]: i have updated

**Pull request merged**